### PR TITLE
SOC-2378

### PR DESCRIPTION
**Pull request by mleczey**
## Links
https://wikia-inc.atlassian.net/browse/SOC-2378

## Descriptions
Automated test for locking posts.

## Reviewers
@tortila

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/ReplyCreatorDesktop.java [hunk 1]
```diff
@@ -1,7 +1,9 @@
 package com.wikia.webdriver.elements.mercury.components.discussions.desktop;
 
+import com.wikia.webdriver.elements.mercury.components.discussions.mobile.ReplyCreatorMobile;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/ReplyCreatorDesktop.java]: MINOR Remove this unused import 'com.wikia.webdriver.elements.mercury.components.discussions.mobile.ReplyCreatorMobile'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DiscussionsConstants.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.common;
+
+public class DiscussionsConstants {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DiscussionsConstants.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1118)

File: src/test/java/com/wikia/webdriver/common/remote/operations/BaseRemoteOperation.java [hunk 3]
```diff
@@ -10,8 +10,10 @@
 import lombok.Getter;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.http.HttpEntity;
 import org.apache.http.client.methods.CloseableHttpResponse;
 import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/remote/operations/BaseRemoteOperation.java]: MINOR Remove this unused import 'org.apache.http.client.methods.HttpEntityEnclosingRequestBase'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/MoreOptionsPopOver.java [hunk 4]
```diff
@@ -1,50 +1,84 @@
...
+import com.google.common.base.Predicate;
+import com.google.common.base.Predicates;
+import lombok.AccessLevel;
 import lombok.AllArgsConstructor;
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.FluentWait;
 
 import java.util.List;
+import java.util.Objects;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/MoreOptionsPopOver.java]: MINOR Remove this unused import 'java.util.Objects'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 3 major
* MINOR 3 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/5e4d434919b3b95cff4114ab733707f68b80d450/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/ReplyCreatorDesktop.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/5e4d434919b3b95cff4114ab733707f68b80d450/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/mobile/ReplyCreatorMobile.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

File: src/test/java/com/wikia/webdriver/testcases/discussions/LockingPostTests.java [hunk 5]
```diff
@@ -0,0 +1,573 @@
...
+import com.wikia.webdriver.elements.mercury.pages.discussions.PostDetailsPage;
+import com.wikia.webdriver.elements.mercury.pages.discussions.PostsListPage;
+import com.wikia.webdriver.elements.mercury.pages.discussions.UserPostsPage;
+import org.testng.annotations.Test;
+
+@Execute(onWikia = MercuryWikis.DISCUSSIONS_AUTO)
+@Test(groups = "discussions-locking-posts")
+public class LockingPostTests extends NewTestTemplate {
+
+  private static final String DESKTOP_RESOLUTION = "1920x1080";
```
Comment (tortila) [inline src/test/java/com/wikia/webdriver/testcases/discussions/LockingPostTests.java]: can you also put `DESKTOP_RESOLUTION` in `DiscussionConstants`?

File: src/test/java/com/wikia/webdriver/testcases/discussions/LockingPostTests.java [hunk 6]
```diff
@@ -0,0 +1,573 @@
...
+  @Test(groups = {"discussions-locking-posts-desktop", "discussions-anonymousUserDesktopLocking"})
+  @Execute(asUser = User.ANONYMOUS)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
+  public void anonymousUserOnDesktopCanNotLockAPostOnUserPostsPage() {
+    final MoreOptionsPopOver moreOptionsPopOver = findMoreOptionsOnUserPostsPage();
+
+    Assertion.assertFalse(moreOptionsPopOver.hasLockPostOption(), String.format(SHOULD_NOT_LOCK_MESSAGE, User.ANONYMOUS.name()));
+  }
+
+  // Second User on mobile
```
Comment (tortila) [inline src/test/java/com/wikia/webdriver/testcases/discussions/LockingPostTests.java]: who - in terms of permissions - us `Second User`?

Comment (mleczey) [inline src/test/java/com/wikia/webdriver/testcases/discussions/LockingPostTests.java]: Normal regular user, but different from user (without number).

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/91728768558c7c3e0e48452bbb584c9dfb4744ba/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/ReplyCreatorDesktop.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/91728768558c7c3e0e48452bbb584c9dfb4744ba/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/mobile/ReplyCreatorMobile.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

**Pull request merged**